src/creature.h

```diff
@@ -235,6 +235,7 @@ class Creature
         // increase pain, apply effects, etc
         virtual void apply_damage(Creature *source, body_part bp, int amount) = 0;
 
+        virtual void on_damage(Creature *source, int amount) = 0; // if this receives actual damage, do something with it
```

Comment:
Don't pass amount as is, pass the whole `damage_dealt` structure instead. Otherwise dealing damage with electricity or fire will spill as much acid as cutting.